sv.c

```diff
@@ -4771,6 +4774,429 @@ Perl_sv_setsv_flags(pTHX_ SV *dsv, SV* ssv, const I32 flags)
+         * for more information. The main point is that the
+         * SVf_POK should not have been set on dsv either and so
+         * we assert that here. */
+        assert(!(SvFLAGS(dsv) & SVf_POK));
+    }
+
+    {
+        const char *vstr_pv;
+        STRLEN vstr_len;
+        if ((vstr_pv = SvVSTRING(ssv, vstr_len))) {
+            sv_magic(dsv, NULL, PERL_MAGIC_vstring, vstr_pv, vstr_len);
+            SvRMAGICAL_on(dsv);
+        }
+    }
+    if (SvTAINTED(ssv))
+        SvTAINT(dsv);
+    return;
+}
+
+/*
+=for apidoc sv_freshcopy_flags
+
+This creates a new SV which contains the values of the original SV.
+
+It does the bulk of the work for C<newSVsv_flags> and C<sv_mortalcopy_flags>.
+Less common cases are passed to C<sv_setsv_flags>.
+
+This function accepts the same flags as C<sv_setsv_flags>, with the single
+addition of C<SVs_TEMP>, which toggles the treatment of a freed source SV:
+  * Not present: emulate C<newSVsv_flags> by emitting a warning and
+                 returning NULL.
+  * Present:     emulate C<sv_mortalcopy_flags> behaviour: C<croak()>.
+
+=cut
+*/
+
+    /* Note: If a PVIV/PVNV/PVMG is only IOK, NOK, ROK, it is _mostly_
+     * possible to create just a headless SV to store that value.
+     * Some parts of core (Perl_amagic_call in gv.c specifically) do
+     * assume - and possibly CPAN might - that SvTYPE(dsv) == SvTYPE(ssv)
+     * though, which is why the code below does not try that type
+     * simplification. Perhaps this might be worth revisiting in the future.
+     * -- April 2025. */
+
+SV *
+Perl_sv_freshcopy_flags(pTHX_ SV *const old, I32 flags)
+{
+    PERL_ARGS_ASSERT_SV_FRESHCOPY_FLAGS;
+    SV *dsv;
+    new_SV(dsv);
+
+    /* new_SV includes default initialization of SvFLAGS and SvANY.
+     * However, the SVt_IV cases in the switch below are both very common
+     * and very simple. If we initialize for those cases, a decent compiler
+     * will hopefully elide new_SVs defaults so no extra work is done.
+     * For all other cases, SvANY and SvFLAGS will be overwritten anyway. */
+    SET_SVANY_FOR_BODYLESS_IV(dsv);
+    SvFLAGS(dsv) = SVt_IV|SVf_IOK|SVp_IOK;
+
+    const U32 sflags = SvFLAGS(old);
+
+    /* These are the hottest and simplest cases */
+    switch( sflags & (SVTYPEMASK|SVf_IOK|SVf_IVisUV|SVf_ROK|SVf_NOK|SVf_POK) ) {
+        case SVt_IV|SVf_IOK: /* [30%] */
+            assert(SvANY(dsv));
+            assert(SvFLAGS(dsv) == (SVt_IV|SVf_IOK|SVp_IOK) );
+
+            assert(    &(old->sv_u.svu_iv)
+                == &(((XPVIV*) SvANY(old))->xiv_iv));
+            assert(    &(dsv->sv_u.svu_iv)
+                == &(((XPVIV*) SvANY(dsv))->xiv_iv));
+            dsv->sv_u.svu_iv = old->sv_u.svu_iv;
+            break;
+        case SVt_IV|SVf_ROK: /* [12%] */
+            assert(SvANY(dsv));
+#if defined (DEBUGGING) || defined (PERL_DEBUG_COW)
+            dsv->sv_u.svu_rv = SvREFCNT_inc(SvRV(old));
+#else
+            dsv->sv_u.svu_rv = SvREFCNT_inc( old->sv_u.svu_rv );
```

Comment:
ADD `SvREFCNT_inc_simple_NN(sv)` its faster

Reply:
I think `SvROK_on()` + `SvPVX() == NULL` is an illegal/invalid combination in the Perl C API, But maybe a 2nd person can confirm its an illegal combo. Note, I can't see the lines above the quote through GH UI. BUT BUT, if the top of this branch is something like
```
if (ssvtype == SVT_IV  && dsvtype == SVT_IV) {
    FLAGS(dsv) = FLAGS(dsv) |  ( FLAGS(ssv) &  (SVf_IV | SVf_IVp | SVf_IsUV | SVf_ROK) );
```
then who knows what on earth, the bytes/bitpattens in `SV` head union aka `SvPVX()`  mean.

I doubt the GH UI quote I am looking at is doing something like what I described, because then a non-0 UV/IV value, would test true as a `SV*` ptr inside `SvREFCNT_inc()` and SEGV instantly.

Reply:
I'll include this suggestion in the next push. ROK + invalid pointer seems like it would be a bug elsewhere.